docs/setup_configuration.rst

```diff
@@ -0,0 +1,92 @@
+==========================
+Django Setup Configuration
+==========================
+
+There is optional support for ``django-setup-configuration`` that allows you to automatically configure the
+OpenID Connect configuration the the ``setup_configuration`` commmand.
+
+You must install the ``setupconfig`` dependency:
+
+
+.. code-block:: bash
+
+    pip install mozilla-django-oidc-db[setupconfig]
+
+
+You must then define the required and any optional django settings mentioned below and
+put the ``AdminOIDCConfigurationStep`` in your django-setup-configuration steps:
+
+.. code-block:: python
+
+    SETUP_CONFIGURATION_STEPS = [
+        ...
+        "mozilla_django_oidc_db.setupconfig.bootstrap.AdminOIDCConfigurationStep",
+        ...
+    ]
+
+Configuration Settings:
+=======================
+
+* ``OIDC_DB_CONFIG_ENABLE``: enable setup configuration step
+
+* ``OIDC_DB_SETUP_CONFIG_ADMIN_AUTH``: Dictionary that maps OIDC fields to their values.
+
+
+Example:
+
+.. code-block:: python
+
+    OIDC_DB_SETUP_CONFIG_ADMIN_AUTH = {
+        "oidc_rp_client_id": "client-id",
+        "oidc_rp_client_secret": "secret",
+        "oidc_op_authorization_endpoint": "https://keycloak.local/protocol/openid-connect/auth",
+        "oidc_op_token_endpoint": "https://keycloak.local/protocol/openid-connect/token",
+        "oidc_op_user_endpoint": "https://keycloak.local/protocol/openid-connect/userinfo",
+    }
+
+
+Required Fields:
+""""""""""""""""
+
+
+* ``oidc_rp_client_id``: OpenID Connect client ID provided by the OIDC Provider.
+* ``oidc_rp_client_secret``: OpenID Connect secret provided by the OIDC Provider.
+
+The discovery endpoint can be provided to automatically fetch the other endpoints. Otherwise they must be provided individually.
+
+* ``oidc_op_discovery_endpoint``: URL of your OpenID Connect provider discovery endpoint ending with a slash (`.well-known/...` will be added automatically).
+
+   **OR**
+
+* ``oidc_op_authorization_endpoint``: URL of your OpenID Connect provider authorization endpoint
+* ``oidc_op_token_endpoint``: URL of your OpenID Connect provider token endpoint
+* ``oidc_op_user_endpoint``: URL of your OpenID Connect provider userinfo endpoint
+
+
+
+Optional Fields:
+""""""""""""""""
```

Comment:
For the future - there seems to be a lot of duplication/repetition of model field help texts. It would be a lot nicer if we can just use Sphinx (custom directive) to extract the help text _and_ default value from the model field definition.

RST:

```rst
* :model_field:`oidc_op_jwks_endpoint`
```

and then the python code for the directive could make use of:

```py
>>> field = OIDCConfig._meta.get_field("oidc_op_jwks_endpoint") 
>>> field.help_text
...
>>> field.default
...
```

Reply:
Made some attempt at this. Added it to a separate PR ( branched form this one)  https://github.com/maykinmedia/mozilla-django-oidc-db/pull/117

We could create a common repo with sphinx roles/directives